extensions/wikia/CommunityPage/models/CommunityPageSpecialUsersModel.class.php

```diff
@@ -106,36 +106,48 @@ function () {
 				self::logUserModelPerformanceData( 'query', 'all_admins' );
 
 				$db = wfGetDB( DB_SLAVE );
-
 				$adminIds = $this->getAdmins();
 
 				if ( !$adminIds ) {
 					return [];
 				}
 
 				$botIds = $this->getBotIds();
-				$dateTwoYearsAgo = date( 'Y-m-d', strtotime( '-2 years' ) );
+
+				$validAdminIds = array_diff( $adminIds, $botIds );
+				$validAdmins = [];
+
+				foreach ( $validAdminIds as $id ) {
+					$validAdmins[ $id ] = [
+						'userId' => $id,
+						'latestRevision' => '',
+						'isAdmin' => true,
+					];
+				}
 
 				$sqlData = ( new WikiaSQL() )
 					->SELECT( 'rev_user_text, rev_user, MAX(rev_timestamp) AS latest_revision' )
 					->FROM ( 'revision FORCE INDEX (user_timestamp)' )
 					->WHERE( 'rev_user' )->NOT_EQUAL_TO( 0 )
-					->AND_( 'rev_user' )->IN( $adminIds )
+					->AND_( 'rev_user' )->IN( $validAdminIds )
 					->AND_( 'rev_user' )->NOT_IN( $botIds )
-					->AND_( 'rev_timestamp' )->GREATER_THAN( $dateTwoYearsAgo )
 					->GROUP_BY( 'rev_user' )
 					->ORDER_BY( 'latest_revision DESC' );
 
-				$result = $sqlData->runLoop( $db, function ( &$result, $row ) {
-					$result[] = [
-						'userId' => $row->rev_user,
-						'latestRevision' => $row->latest_revision,
-						'isAdmin' => true,
-					];
+				$sqlData->runLoop( $db, function ( &$result, $row ) use ( &$validAdmins ) {
+					$validAdmins[$row->rev_user]['latestRevision'] = $row->latest_revision;
 				} );
 
-				return $result;
-			}
+				uasort( $validAdmins, function( $a, $b ) {
+					if ( $a[ 'latestRevision' ] === $b[ 'latestRevision' ] ) {
```

Comment:
Do wee need this comparison? Maybe we could also extract this sorting to separate method?